core/src/test/java/org/apache/iceberg/TestManifestGroupPlanFiles.java

```diff
@@ -0,0 +1,67 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.iceberg;
+
+import java.io.IOException;
+import java.lang.reflect.Field;
+import java.util.concurrent.ConcurrentLinkedQueue;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import org.apache.iceberg.io.CloseableIterator;
+import org.apache.iceberg.relocated.com.google.common.collect.ImmutableList;
+import org.junit.Assert;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.junit.runners.Parameterized;
+
+@RunWith(Parameterized.class)
+public class TestManifestGroupPlanFiles extends TestManifestReader {
+  @Parameterized.Parameters(name = "formatVersion = {0}")
+  public static Object[] parameters() {
+    return new Object[] {1, 2};
+  }
+
+  public TestManifestGroupPlanFiles(int formatVersion) {
+    super(formatVersion);
+  }
+
+  @Test
+  public void testCloseParallelIteratorWithoutCompleteIteration()
```

Comment:
This is a good idea. At first, I thought about `TestParallelIterable `, but I couldn't find `TestCloseableIterable` as reference. 
Because I don't know much about the iceberg test specification, so I just wrote a test referring to `TestManifestReader`.

Reply:
it's here: https://github.com/apache/iceberg/blob/master/api/src/test/java/org/apache/iceberg/io/TestCloseableIterable.java#L38. Given that `ParallelIterable` is in the `iceberg-core` module under `org.apache.iceberg.util` you can just add `TestParallelIterable` under `org.apache.iceberg.util` in the `iceberg-core` test module

Reply:
@nastra I push new commit, just like your suggestion.